app.py

```diff
@@ -2,15 +2,35 @@
 import numpy as np
 from kmeans.parallel import ParallelKMeans
 from kmeans.sequential import SequentialKMeans
+import csv
 
 np.random.seed(1234)
 
 K = 3
-N = 10000
 M = 2
 max_iter = 10
 
-X = np.random.rand(N, M)
+def load_data(filename):
+    data = []
+    with open(filename, 'r') as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            data.append([float(x) for x in row])
+    return np.array(data)
+
+def count_csv_rows(csv_file):
+    row_count = 0
+    with open(csv_file, 'r') as file:
+        reader = csv.reader(file)
+        for row in reader:
+            row_count += 1
+    return row_count
+
+filename = 'data_1000.csv'
```

Comment:
You can't run the program if you don't have the CSV file. Gonna be a bad experience in fresh runs. Re write this flow

1. Check if the csv exists
2. If file exists, load it
3. Else, generate the csv, save it in disk and load it